api/src/public/login/script.js

```diff
@@ -124,12 +125,37 @@ const setRedirectUrl = function() {
   }
 };
 
+const checkSession = function() {
+  if (getCookie('login') === 'force') {
+    // require user to login regardless of session state
+    return;
+  }
+  const redirect = encodeURIComponent(document.getElementById('redirect').value);
+  request('GET', '/medic/login/identity?redirect=' + redirect, null, function(xmlhttp) {
+    if (xmlhttp.status === 0 || xmlhttp.status === 401) {
+      // no internet connection or not logged in - ignore
+      return;
+    }
+    if (xmlhttp.status !== 200) {
+      return console.error(`Could not determine session status. Response: ${xmlhttp.status}, ${xmlhttp.response}`);
+    }
+    try {
+      const response = JSON.parse(xmlhttp.response);
+      window.location = response.url || '/';
+    } catch (e) {
+      return console.error('Could not parse session status.', e);
+    }
+  });
+};
+
 document.addEventListener('DOMContentLoaded', function() {
+  setRedirectUrl();
+  checkSession();
```

Comment:
There's one disadvantage to this approach. If i load the login while being logged in and have a poor connection, I'll see the login page until this request passes (with network throttled to "slow 3G" it's a 4-5 s duration). 

Also, if I set my browser connection to be offline and visit the login page, even if I'm logged in, I'm not being redirected to the app. This is no different from what is happening now, though - because the login page is actually being served from SW cache and it's not even trying to reach API to check login status - so not a regression.

Reply:
Yes, you're right. The problem is we can't actually check if the user is logged in without an internet connection as their session may have timed out. Similarly we can't tell if they are connected to the internet in an offline first way.

The only way I can think of is by checking the `userCtx` cookie. If they have the cookie then we could assume they're logged in and redirect them to the app, then if the session is expired and they have a connection the app will kick them back to the login page once the server request comes back. The code to send them to the login page also deletes the `userCtx` cookie so that should break the infinite loop, however I'm concerned that it may be too fragile. 

We could include a query param when sending them to the login page as a failsafe? Alternatively we could check the `document.referrer` to see where they came from and stop the redirect that way? Do you have any other ideas?